api/v1/controllers/lenses.js

```diff
@@ -410,18 +411,46 @@ module.exports = {
 
       updateLensDetails(seqObj);
       const assocToCreate = u.includeAssocToCreate(seqObj, helper);
-      helper.model.create(seqObj, assocToCreate)
-      .then((o) => {
-        resultObj.dbTime = new Date() - resultObj.reqStartTime;
-        delete o.dataValues.library;
-        u.logAPI(req, resultObj, o.dataValues);
-        res.status(httpStatus.CREATED).json(
-          u.responsify(o, helper, req.method)
-        );
-      })
-      .catch((err) => {
-        u.handleError(next, err, helper.modelName);
-      });
+      if (!featureToggles.isFeatureEnabled('returnCreatedBy')) {
+        createLens();
+      } else {
+        authUtils.getUser(req)
+        .then((user) => {
+          if (user) {
+            seqObj.createdBy = user.id;
+          }
+
+          return createLens();
+        })
+        .catch((err) => {
+          if (err.status === 403) {
+            return createLens();
+          } else {
+            u.handleError(next, err, helper.modelName);
+          }
+        });
+      }
+
+      function createLens() {
+        return helper.model.create(seqObj, assocToCreate)
+        .then((o) => {
+          resultObj.dbTime = new Date() - resultObj.reqStartTime;
+          delete o.dataValues.library;
+          u.logAPI(req, resultObj, o.dataValues);
+          if (featureToggles.isFeatureEnabled('returnCreatedBy')) {
+            o.reload()
```

Comment:
the associations are not loaded on POST response, so there is no user object returned. Another example is samples response: it does not have aspects object on POST response. So we need to reload the instance to get the associations.